### PR TITLE
Shipping Labels: handle the value selected in Carrier and Rates

### PR DESCRIPTION
**Pull request by pmusolino**
Part of #4085 

## Description
As part of Shipping Labels, I managed the value selected in Carrier and Rates, and now is also possible to re-open the Carrier and Rates screen for selecting a new rate.
In this PR:
- Added the Done button in Carrier and Rates screen. It will be enabled only if there is a rate selected.
- Pressing the Done button in Carrier and Rates screen the rate selected will be shown under the "Carrier and Rates" cell in "Create Shipping Label" view.
- Now it's possible to re-open the Carrier and Rates screen with the previous rate selected, and change it.

## Testing
1. Launch the app in debug mode (because of the feature flag).
2. Open an order detail that is eligible for the shipping label.
3. Press the button "Create Shipping Label".
4. Press the button "Continue" in Create Shipping Label form, under the **Ship From** cell. Confirm the address.
5. Press the button "Continue" in Create Shipping Label form, under the **Ship To** cell. Confirm the address.
6. Press the button "Continue" in Create Shipping Label form, under the **Package Details** cell.
7. You should see the screen that shows the Package Details screen and all the items that will be shipped.
8. Press the "Package selected" row. 
9. Select a package, then press "Done".
10. Add a Total package weight, press "Done".
11. The "Shipping Carrier and Rates" cell should be enabled now. Press it.
12. You should see the new screen "Carrier and Rates". It should display first the ghost cells, then the data fetched from the API.
13. Select a carrier and press Done.
14. The info of the selected carrier and the rate should be shown on the previous screen.
15. You should be able to change the selected rate.

| Cell Shipping Carrier and Rates with info showed            |  Selected carrier and rate |
:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone 12 Pro - 2021-05-26 at 18 12 25 | Simulator Screen Shot - iPhone 12 Pro - 2021-05-26 at 18 12 34

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Carriers and Rates/ShippingLabelCarriers.swift [hunk 1]
```diff
@@ -5,9 +5,21 @@ import Shimmer
...
     @ObservedObject private var viewModel: ShippingLabelCarriersViewModel
+    @Environment(\.presentationMode) var presentation
 
-    init(viewModel: ShippingLabelCarriersViewModel) {
+    /// Completion callback
+    ///
+    typealias Completion = (_ selectedRate: ShippingLabelCarrierRate?,
+
+                            _ selectedSignatureRate: ShippingLabelCarrierRate?,
+
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Carriers and Rates/ShippingLabelCarriers.swift:15]: nit: unnecessary newlines?

File: WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Carriers and Rates/ShippingLabelCarriers.swift [hunk 2]
```diff
@@ -44,6 +65,7 @@ private extension ShippingLabelCarriers {
         static let titleView = NSLocalizedString("Carrier and Rates",
```
Comment (ealeksandrov) [inline WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Carriers and Rates/ShippingLabelCarriers.swift:66]: nit: unnecessary newline

**Review by ealeksandrov — APPROVED** [review]
LGTM :shipit:

While testing I once saw confusing empty screen after loading state. I wasn't able to reproduce it but I think error handling is planned later anyway.

**Comment by pmusolino** [comment]
Thank you for your review @ealeksandrov!

> While testing I once saw confusing empty screen after loading state. I wasn't able to reproduce it but I think error handling is planned later anyway. 

Yes, I've in the plan to work on it today 😉  it will be part of another PR.

**Pull request merged**